HandBrake build fails without hardware acceleration enabled

### Problem Description

> ./configure --launch-jobs=$(nproc) --launch

Results in:
```
/usr/lib64/ccache/g++ -pipe -fmessage-length=0 -Wall -Wno-format-truncation -g3 -O3 -mfpmath=sse -msse2 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I./libhb/ -I./contrib/include -o HandBrakeCLI -Wl,--start-group test/parsecsv.o test/test.o ./libhb/libhandbrake.a -L./contrib/lib -lass -lavformat -lavfilter -lavcodec -lavutil -lswresample -lmp3lame -ldvdnav -ldvdread -lfribidi -lswscale -lvpx -ltheoraenc -ltheoradec -lvorbis -lvorbisenc -logg -lx264 -lbluray -lfreetype -lbz2 -lz -ljansson -lharfbuzz -lopus -lspeex -llzma -ldav1d -lturbojpeg -lzimg -lSvtAv1Enc -lfontconfig -lx265 -ldovi -lpthread -ldl -lm -lnuma -Wl,--end-group
/usr/bin/ld: ./contrib/lib/libavutil.a(hwcontext_drm.o): in function `drm_device_create':
/home/jstebbins/Source/hb/HandBrake/build/contrib/ffmpeg/ffmpeg-7.0/libavutil/hwcontext_drm.c:60: undefined reference to `drmGetVersion'
/usr/bin/ld: /home/jstebbins/Source/hb/HandBrake/build/contrib/ffmpeg/ffmpeg-7.0/libavutil/hwcontext_drm.c:73: undefined reference to `drmFreeVersion'
collect2: error: ld returned 1 exit status
make: *** [../test/module.rules:49: HandBrakeCLI] Error 1
```
It appears we are enabling nvenc in ffmpeg even though it is not enabled in the HandBrake configure command. The above error is due to libdrm not being included in the final link (required by nvenc encoders).

ffmpeg enabled encoders
```
Enabled encoders:
aac                     ffv1                    hevc_nvenc              libvpx_vp9
ac3                     flac                    libmp3lame              mpeg2video
av1_nvenc               h263                    libopus                 mpeg4
eac3                    h264_nvenc              libvpx_vp8              truehd

```

### Activity Log, Crash Log or any other details

```text
NA
```


### What Operating System are you running?

Linux

### What version of HandBrake are you running?

git HEAD

### Where did you download HandBrake from?

NA

--- a/make/include/gcc.defs
+++ b/make/include/gcc.defs
@@ -87,6 +87,7 @@ GCC.args.F                 = -F$(1)
 GCC.args.f                 = -framework $(1)
 GCC.args.L                 = -L$(1)
 GCC.args.l                 = -l$(1)
+GCC.args.pkgconfig         = $(1)
 GCC.args.end               = -Wl,--end-group
 
 GCC.args.extra         =
@@ -188,6 +189,7 @@ define import.GCC
     $(1).GCC.args.f         = $$(GCC.args.f)
     $(1).GCC.args.L         = $$(GCC.args.L)
     $(1).GCC.args.l         = $$(GCC.args.l)
+    $(1).GCC.args.pkgconfig = $$(GCC.args.pkgconfig)
     $(1).GCC.args.end       = $$(GCC.args.end)
 
     $(1).GCC.args.extra         = $$(GCC.args.extra)
@@ -226,18 +228,18 @@ define import.GCC
     $(1).GCC.i = $$(4)
 
     # FUNCTION: C link dynamic-lib
-    $(1).GCC.DYLIB.args = !gcc ?c_std ?pipe ?strip ?dylib ?extra.dylib ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *i !a ?end
+    $(1).GCC.DYLIB.args = !gcc ?c_std ?pipe ?strip ?dylib ?extra.dylib ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
     $(1).GCC.DYLIB = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.DYLIB.args),$$(1),$$(2))
 
     # FUNCTION: C link executable
-    $(1).GCC.EXE.args = !gcc ?c_std ?pipe ?strip ?extra.exe ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *i !a ?end
+    $(1).GCC.EXE.args = !gcc ?c_std ?pipe ?strip ?extra.exe ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
     $(1).GCC.EXE = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.EXE.args),$$(1),$$(2))
 
     # FUNCTION: C++ link dynamic-lib
-    $(1).GCC.DYLIB++.args = !gxx ?cxx_std ?pipe ?strip ?dylib ?extra.dylib++ ?ML *W *archs *sysroot *minvers ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *i !a ?end
+    $(1).GCC.DYLIB++.args = !gxx ?cxx_std ?pipe ?strip ?dylib ?extra.dylib++ ?ML *W *archs *sysroot *minvers ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
     $(1).GCC.DYLIB++ = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.DYLIB++.args),$$(1),$$(2))
 
     # FUNCTION: C++ link executable
-    $(1).GCC.EXE++.args = !gxx ?cxx_std ?pipe ?strip ?extra.exe++ ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *i !a ?end
+    $(1).GCC.EXE++.args = !gxx ?cxx_std ?pipe ?strip ?extra.exe++ ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
     $(1).GCC.EXE++ = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.EXE++.args),$$(1),$$(2))
 endef

--- a/make/include/gcc.defs
+++ b/make/include/gcc.defs
@@ -228,18 +228,18 @@ define import.GCC
     $(1).GCC.i = $$(4)
 
     # FUNCTION: C link dynamic-lib
-    $(1).GCC.DYLIB.args = !gcc ?c_std ?pipe ?strip ?dylib ?extra.dylib ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
+    $(1).GCC.DYLIB.args = !gcc ?c_std ?pipe ?strip ?dylib ?extra.dylib ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l !pkgconfig *i !a ?end
     $(1).GCC.DYLIB = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.DYLIB.args),$$(1),$$(2))
 
     # FUNCTION: C link executable
-    $(1).GCC.EXE.args = !gcc ?c_std ?pipe ?strip ?extra.exe ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
+    $(1).GCC.EXE.args = !gcc ?c_std ?pipe ?strip ?extra.exe ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l !pkgconfig *i !a ?end
     $(1).GCC.EXE = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.EXE.args),$$(1),$$(2))
 
     # FUNCTION: C++ link dynamic-lib
-    $(1).GCC.DYLIB++.args = !gxx ?cxx_std ?pipe ?strip ?dylib ?extra.dylib++ ?ML *W *archs *sysroot *minvers ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
+    $(1).GCC.DYLIB++.args = !gxx ?cxx_std ?pipe ?strip ?dylib ?extra.dylib++ ?ML *W *archs *sysroot *minvers ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l !pkgconfig *i !a ?end
     $(1).GCC.DYLIB++ = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.DYLIB++.args),$$(1),$$(2))
 
     # FUNCTION: C++ link executable
-    $(1).GCC.EXE++.args = !gxx ?cxx_std ?pipe ?strip ?extra.exe++ ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l *pkgconfig *i !a ?end
+    $(1).GCC.EXE++.args = !gxx ?cxx_std ?pipe ?strip ?extra.exe++ ?ML *W *archs *sysroot *minver ?vis ?pic .g .O ?extra .lto *D *I !o ?muldefs ?start !i *F *f *L *l !pkgconfig *i !a ?end
     $(1).GCC.EXE++ = $$(call fn.ARGS,$(1).GCC,$$($(1).GCC.EXE++.args),$$(1),$$(2))
 endef

--- a/test/module.defs
+++ b/test/module.defs
@@ -1,6 +1,9 @@
 $(eval $(call import.MODULE.defs,TEST,test,LIBHB))
 $(eval $(call import.GCC,TEST))
 
+TEST.PKGCONFIG.env = PKG_CONFIG_PATH="$(call fn.ABSOLUTE,$(CONTRIB.build/))lib/pkgconfig:$(PKG_CONFIG_PATH)"
+TEST.PKGCONFIG.exe = $(TEST.PKGCONFIG.env) $(PKGCONFIG.exe)
+
 TEST.src/   = $(SRC/)test/
 TEST.build/ = $(BUILD/)test/
 
@@ -13,26 +16,21 @@ TEST.GCC.L = $(CONTRIB.build/)lib
 
 TEST.libs = $(LIBHB.a)
 
-TEST.GCC.l = \
-        ass avformat avfilter avcodec avutil swresample mp3lame dvdnav \
-        dvdread fribidi swscale vpx theoraenc theoradec vorbis vorbisenc ogg \
-        x264 bluray freetype bz2 z jansson harfbuzz opus speex lzma dav1d \
-        turbojpeg zimg SvtAv1Enc
+TEST.pkgconfig_libs = libass libavformat libavfilter libavcodec  libavutil libswresample dvdnav \
+	dvdread fribidi libswscale vpx theoraenc theoradec vorbis vorbisenc ogg x264 libbluray freetype2 \
+	bzip2 zlib jansson harfbuzz opus speex liblzma dav1d libturbojpeg SvtAv1Enc
 
 ifeq (,$(filter $(HOST.system),darwin cygwin mingw))
-    TEST.GCC.l += fontconfig
-endif
-
-ifeq (1,$(FEATURE.qsv))
-    TEST.GCC.l += vpl
-ifneq (,$(filter $(HOST.system),linux freebsd))
-    TEST.GCC.l += va va-drm drm
-endif
+    TEST.pkgconfig_libs += fontconfig
 endif
 
 ifeq (1,$(FEATURE.flatpak))
-    TEST.GCC.l += glib-2.0
+    TEST.pkgconfig_libs += glib-2.0
 endif
+
+TEST.GCC.pkgconfig = $(shell $(TEST.PKGCONFIG.exe) --libs $(TEST.pkgconfig_libs))
+
+TEST.GCC.l = mp3lame zimg
 
 TEST.GCC.l += $(foreach m,$(MODULES.NAMES),$($m.OSL.libs))
 

--- a/test/module.defs
+++ b/test/module.defs
@@ -16,23 +16,17 @@ TEST.GCC.L = $(CONTRIB.build/)lib
 
 TEST.libs = $(LIBHB.a)
 
-TEST.pkgconfig_libs = libass libavformat libavfilter libavcodec  libavutil libswresample dvdnav \
-	dvdread fribidi libswscale vpx theoraenc theoradec vorbis vorbisenc ogg x264 libbluray freetype2 \
-	bzip2 zlib jansson harfbuzz opus speex liblzma dav1d libturbojpeg SvtAv1Enc
+TEST.pkgconfig_libs = libass libavformat libavfilter libavcodec libavutil libswresample dvdnav \
+	dvdread libswscale theoraenc theoradec vorbis vorbisenc ogg x264 libbluray \
+	jansson libturbojpeg SvtAv1Enc
 
-ifeq (,$(filter $(HOST.system),darwin cygwin mingw))
-    TEST.pkgconfig_libs += fontconfig
-endif
+TEST.pkgconfig_libs += $(foreach m,$(MODULES.NAMES),$($m.OSL.libs))
 
 ifeq (1,$(FEATURE.flatpak))
     TEST.pkgconfig_libs += glib-2.0
 endif
 
 TEST.GCC.pkgconfig = $(shell $(TEST.PKGCONFIG.exe) --libs $(TEST.pkgconfig_libs))
-
-TEST.GCC.l = mp3lame zimg
-
-TEST.GCC.l += $(foreach m,$(MODULES.NAMES),$($m.OSL.libs))
 
 TEST.install.exe = $(DESTDIR)$(PREFIX/)bin/$(notdir $(TEST.exe))
 ifeq (1,$(FEATURE.flatpak))

